elasticsearch-evolution-core/src/main/java/com/senacor/elasticsearch/evolution/core/internal/migration/input/MigrationScriptReaderImpl.java

```diff
@@ -33,9 +44,128 @@ public MigrationScriptReaderImpl(List<String> locations,
         this.esMigrationSuffixes = esMigrationFileSuffixes;
     }
 
-    @Override
+    /**
+     * Reads all migration scripts from the specified locations that match prefixes and suffixes
+     *
+     * @return a list of {@link RawMigrationScript}
+     */
     public List<RawMigrationScript> read() {
-        // TODO (ak) impl
-        return Collections.emptyList();
+        List<RawMigrationScript> migrationScripts = new ArrayList<>();
+
+        this.locations.stream()
+                .map(location -> {
+                    try {
+                        return readFromLocation(location);
```

Comment:
Die Exception könntest du auch in der Methode schon umwandeln, dann könntest du hier einfach eine methoden referentz übergeben.